Compilation with gpgsql module: cannot find -lnsl

I'm trying compiling powerdns with gpgsql module for PostreSQL on Alpine linux (Docker) and I get this following error:

```bash
CXXLD  pdns_server
/usr/lib/gcc/x86_64-alpine-linux-musl/5.3.0/../../../../x86_64-alpine-linux-musl/bin/ld: cannot find -lnsl
collect2: error: ld returned 1 exit status
Makefile:1379: recipe for target 'pdnssec' failed
make[4]: *** [pdnssec] Error 1
make[4]: *** Waiting for unfinished jobs....
/usr/lib/gcc/x86_64-alpine-linux-musl/5.3.0/../../../../x86_64-alpine-linux-musl/bin/ld: cannot find -lnsl
```

What is the lib `nsl`?

You can see the related PR here: https://github.com/dockervan/powerdns/pull/2

And the failing Travis job: https://travis-ci.org/dockervan/powerdns/jobs/102682305#L589-L590

--- a/.travis.yml
+++ b/.travis.yml
@@ -4,6 +4,7 @@ compiler:
   - clang
 before_script:
  - git describe --always --dirty=+
+ - sudo sysctl net.ipv6.conf.lo.disable_ipv6=0
  - sudo /sbin/ip addr add 10.0.3.0/24 dev lo
  - sudo /sbin/ip addr add 1.2.3.4/32 dev lo
  - sudo rm /etc/apt/sources.list.d/travis_ci_zeromq3-source.list

--- a/configure.ac
+++ b/configure.ac
@@ -49,6 +49,10 @@ AC_CHECK_FUNC([gethostent], [], [
  AC_CHECK_LIB([nsl], [gethostent], [LIBS="-lnsl $LIBS"])
 ])
 
+AC_CHECK_FUNC([inet_aton], [], [
+ AC_CHECK_LIB([resolv], [inet_aton], [LIBS="-lresolv $LIBS"])
+])
+
 m4_ifndef([LT_INIT],[
 AC_PROG_LIBTOOL([disable-static])
 ],[

--- a/configure.ac
+++ b/configure.ac
@@ -41,17 +41,9 @@ AS_IF([test "x$enable_hardening" != "xno"], [
   AC_LD_RELRO
 ])
 
-AC_CHECK_FUNC([socket], [], [
-  AC_CHECK_LIB([socket], [socket], [LIBS="-lsocket $LIBS"])
-])
-
-AC_CHECK_FUNC([gethostent], [], [
-  AC_CHECK_LIB([nsl], [gethostent], [LIBS="-lnsl $LIBS"])
-])
-
-AC_CHECK_FUNC([inet_aton], [], [
-  AC_CHECK_LIB([resolv], [inet_aton], [LIBS="-lresolv $LIBS"])
-])
+AC_SEARCH_LIBS([socket], [socket])
+AC_SEARCH_LIBS([gethostent], [nsl])
+AC_SEARCH_LIBS([inet_aton], [resolv])
 
 m4_ifndef([LT_INIT],[
 AC_PROG_LIBTOOL([disable-static])

--- a/configure.ac
+++ b/configure.ac
@@ -164,21 +164,20 @@ AM_CONDITIONAL([OS_MACOSX], false)
 case "$host_os" in
 solaris2.10)
   AC_DEFINE(HAVE_IPV6,1,[If the host operating system understands IPv6])
-  LIBS="-lposix4 -lresolv -lnsl -lsocket -lpthread -lrt $LIBS"
+  LIBS="-lposix4 -lpthread -lrt $LIBS"
   CXXFLAGS="-D_REENTRANT $CXXFLAGS"
   ;;
-
 solaris2.8 | solaris2.9 )
   AC_DEFINE(NEED_POSIX_TYPEDEF,,[If POSIX typedefs need to be defined])
   AC_DEFINE(NEED_INET_NTOP_PROTO,,[If your OS is so broken that it needs an additional prototype])
   AC_DEFINE(HAVE_IPV6,1,[If the host operating system understands IPv6])
-  LIBS="-lposix4 -lresolv -lnsl -lsocket -lpthread $LIBS"
+  LIBS="-lposix4 -lpthread $LIBS"
   CXXFLAGS="-D_REENTRANT $CXXFLAGS"
   ;;
 solaris2.6 | solaris2.7)
   AC_DEFINE(NEED_POSIX_TYPEDEF,,[If POSIX typedefs need to be defined])
   AC_DEFINE(NEED_INET_NTOP_PROTO,,[If your OS is so broken that it needs an additional prototype])
-  LIBS="-lposix4 -lresolv -lnsl -lsocket -lpthread $LIBS"
+  LIBS="-lposix4 -lpthread $LIBS"
   CXXFLAGS="-D_REENTRANT $CXXFLAGS"
   ;;
 linux*)
@@ -332,17 +331,6 @@ for a in $modules; do
 
   if test ${a} = "gpgsql"; then
     LIBS="$LIBS $LIBCRYPT"
-
-    case "$host_os" in
-      freebsd*)
-        ;;
-      darwin*)
-        modulelibs="$modulelibs -lresolv"
-        ;;
-      *)
-        modulelibs="$modulelibs -lresolv -lnsl"
-        ;;
-    esac
   fi
 done
 

--- a/configure.ac
+++ b/configure.ac
@@ -42,15 +42,15 @@ AS_IF([test "x$enable_hardening" != "xno"], [
 ])
 
 AC_CHECK_FUNC([socket], [], [
- AC_CHECK_LIB([socket], [socket], [LIBS="-lsocket $LIBS"])
+  AC_CHECK_LIB([socket], [socket], [LIBS="-lsocket $LIBS"])
 ])
 
 AC_CHECK_FUNC([gethostent], [], [
- AC_CHECK_LIB([nsl], [gethostent], [LIBS="-lnsl $LIBS"])
+  AC_CHECK_LIB([nsl], [gethostent], [LIBS="-lnsl $LIBS"])
 ])
 
 AC_CHECK_FUNC([inet_aton], [], [
- AC_CHECK_LIB([resolv], [inet_aton], [LIBS="-lresolv $LIBS"])
+  AC_CHECK_LIB([resolv], [inet_aton], [LIBS="-lresolv $LIBS"])
 ])
 
 m4_ifndef([LT_INIT],[

